### DISCUSSION TITLE
Enable/disable mask in realtime

### DISCUSSION THREAD

**Discussion by pickonedev** [blakeblackshear/frigate#3517] [General]
Hi!

There is any way in order to disable or enable a mask in frigate?
I want to make an automation, when a person it is detected, mask 0 should be disabled, when no person detected, mask 0 should be enabled.

Thank you in advance!

**Comment by NickM-27** [comment]
This is not possible, can you describe a bit more the usecase? Not sure I understand a situation where this would be necessary. 

Also, what type of mask are you talking about (motion or object)?

**Comment by pickonedev** [comment]
The real situation is that I have a screen projection wall and two doors (left and right of the screen), when the video projector it is off, the mask should be disabled and detect real humans. When the video projector it is on, the mask should be enabled to not detect the humans from the video. But still, this is only a situation, with an option like this, I could integrate with many ideas... I am thinking of any kind of mask, motion or object, does not matter, any of these two :-)
See the image for a real perspective of the situation. The mask should be around the screen

![image](https://user-images.githubusercontent.com/34201387/180646179-2825fc73-38ff-40af-88fc-bbc02e0dc637.png)

**Reply by NickM-27** [reply]
Using a zone I don't think this would be needed, a zone could certainly capture people walking in the room and never a person seen on the screen

**Reply by pickonedev** [reply]
I just started a movie the frigate is seeing all the people from the screen....

**Reply by NickM-27** [reply]
> I just started a movie the frigate is seeing all the people from the screen....

Have you created a zone? It shouldn't matter if frigate sees people on the screen if an event isn't created

Also if you added an object mask on the screen only, it wouldn't block out real people

One critical thing you may not know is that an objects presence in a zone or being masked by an object mask is based on the bottom center of the objects bounding box, not how much it's overlapping. So either a zone on the floor or a constant object mask would work great here as is.

**Reply by pickonedev** [reply]
hmmmm, I didn't thought about that. I thought that frigate will only detect if full person is in the image. I will try, thanks!

**Comment by ediguidedog** [comment]
If you create your zone on the floor and make it a required_zone then people on screen shouldn't trigger as their bounding box wont go much below the screen. Any real person in the room will be spotted

**Reply by pickonedev** [reply]
This is what I am using as a config for frigate:

```
camera_living_2_frigate:
    ffmpeg:
      inputs:
        - path: rtsp://admin:admin@192.168.0.62:8554/Streaming/Channels/101
          roles:
            - detect
    detect:
      width: 1920
      height: 1080
      fps: 5
      enabled: True
    objects:
      track:
        - person
```

How can I achieve this "required_zone", because I didn't know about this option. I think I am aware of what it could do, by the name, but I don't know how to integrate into the config file...
I use this camera only to detect humans, no spanshot, no record, clips, nothing... then all the automations are made in HA

**Reply by NickM-27** [reply]
> How can I achieve this "required_zone", because I didn't know about this option. I think I am aware of what it could do, by the name, but I don't know how to integrate into the config file...
> 
> I use this camera only to detect humans, no spanshot, no record, clips, nothing... then all the automations are made in HA

Create a zone on the floor using frigate debug zone editor and you can use the HA MQTT topic automations to filter `entered_zones` for that zone name to only run automations in that case. 

Another option would be to just cover the screen with a person mask, which similarly wouldn't block out real people walking

**Reply by pickonedev** [reply]
This is what I was thinking after you told me that frigate does not need full body to know that is a real person. Thanks!

**Reply by ediguidedog** [reply]
> Another option would be to just cover the screen with a person mask, which similarly wouldn't block out real people walking

It might if they were on stilts or had the use of a small hot air balloon. :)

**Comment by promufa** [comment]
@blakeblackshear / @NickM-27 first of all thanks for the amazing job you have done with Frigate and all the support you are providing to  the community!

 I was looking for something similar and I am afraid the normal masks / regions would not work in my use case. 
I am using Frigate to automate the stairwell (barely seen on the right, behind the projector screen) lights when a person is detected in the brown-ish zone. 

![image](https://github.com/blakeblackshear/frigate/assets/57601488/a4249fbc-43d7-4ba0-9449-8a0ac7b3f477)

If i create a required region or motion mask over the screen then it mostly overlps the required region that already exists.
For me however, there is no need for this to be realtime. The idea is to apply this mask when the projector is ON (and screen is down). The light would then fall back to a motion sensor that is further up the stairs. Motion sensors are a bit problematic for me because my cats are trigeering them, so they need to be strategically placed to avoid "false" positives. Frigate has been amazing at automating light on person_occupancy and ignore cats moving around.

I guess the question is is there a way to have an automation that changes the frigate.yml entry, add an additional mask and restart frigate ?

**Reply by promufa** [reply]
ooh, I should probably mention that I am running the Frigate HA add-on on a dedictated HA OS x64 machine.

**Reply by NickM-27** [reply]
frigate has apis to edit the config https://github.com/blakeblackshear/frigate/blob/0223d6df60d62b965199c63abfaa72dbee1e77b7/frigate/api/app.py#L274

**Reply by promufa** [reply]
Thanks Nick - my level is not quite there yet but knowing that this is possible in principle is extra motivation to learn some new things :)
THank again for all your work!
